### ISSUE TITLE
Terraform is unable to apply trigger with the change in recipients with different types

### ISSUE THREAD

**Issue by aelmekeev** [honeycombio/terraform-provider-honeycombio#200]
**Versions**
0.8.0

**Steps to reproduce**

1. Start with
```hcl
data "honeycombio_query_specification" "example" {
  calculation {
    op = "COUNT"
  }

  filter {
    column = "trace.parent_id"
    op     = "does-not-exist"
  }
}

resource "honeycombio_query" "example" {
  dataset    = local.dataset
  query_json = data.honeycombio_query_specification.example.json
}

data "honeycombio_recipient" "example" {
  type    = "slack"

  detail_filter {
    name  = "channel"
    value = "#alex-test"
  }
}

data "honeycombio_recipient" "example_slack" {
  type    = "slack"

  detail_filter {
    name  = "channel"
    value = "#test_channel"
  }
}

data "honeycombio_recipient" "example_pd" {
  type    = "pagerduty"

  detail_filter {
    name  = "integration_name"
    value = "Alex Test1"
  }
}

resource "honeycombio_trigger" "example" {
  name        = "Alex test"
  description = "Alex test"

  query_id = honeycombio_query.example.id
  dataset  = local.dataset

  frequency = 1800

  threshold {
    op    = ">"
    value = 1000000000
  }

  recipient {
    id = data.honeycombio_recipient.example.id
  }

  recipient {
    id = data.honeycombio_recipient.example_slack.id
  }

  recipient {
    id = data.honeycombio_recipient.example_pd.id

    notification_details {
      pagerduty_severity = "critical"
    }
  }
}
```

```
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

2. Comment out second recipient and apply:

```hcl
  # recipient {
  #   id = data.honeycombio_recipient.example_slack.id
  # }
```

```
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

3. Run apply again:

We would expect to have 0 changes but... 🐛 

```
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

4. Uncomment the previously commented block and run apply again 💣 

```hcl
honeycombio_trigger.example: Modifying... [id=pVVZZxandMW]
╷
│ Error: 400 Bad Request: PagerDuty target not found
│ 
│   with honeycombio_trigger.example,
│   on test.tf line 139, in resource "honeycombio_trigger" "example":
│  139: resource "honeycombio_trigger" "example" {
│ 
```

**Additional context**

**Comment by aelmekeev** [comment]
0.7.0 is also broken actually but in a slightly different way. Please let me know if I need to raise a separate issue for this.

1. Let's start with this:

```hcl
data "honeycombio_query_specification" "example" {
  calculation {
    op = "COUNT"
  }

  filter {
    column = "trace.parent_id"
    op     = "does-not-exist"
  }
}

resource "honeycombio_query" "example" {
  dataset    = local.dataset
  query_json = data.honeycombio_query_specification.example.json
}

resource "honeycombio_trigger" "example" {
  name        = "Alex test"
  description = "Alex test"

  query_id = honeycombio_query.example.id
  dataset  = local.dataset

  frequency = 1800

  threshold {
    op    = ">"
    value = 1000000000
  }

  recipient {
    type   = "slack"
    target = "#alex-test"
  }

  recipient {
    type   = "slack"
    target = "#test_channel"
  }

  recipient {
    id = "hgvpnCY1vwY"
  }
}
```

```
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

2. Comment second recipient and run apply

```hcl
  # recipient {
  #   type   = "slack"
  #   target = "#test_channel"
  # }
```hcl

```
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

3. Apply again

We would expect to have 0 changes but... 🐛

```
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

4. Uncomment and apply

```hcl
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

And you can do this forever you will never get 0 changes 🤷

**Comment by jharley** [comment]
Fix released in v0.15.0
